src/tests/BuildWasmApps/Wasm.Build.Tests/CommandBase.cs

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Threading.Tasks;
+
+#nullable enable
+
+namespace Wasm.Build.Tests
+{
+    public class Command
+    {
+        private StringBuilder _extraArgsBuilder = new();
+        private string _label;
+
+        protected string _command;
+
+        public Process? CurrentProcess { get; private set; }
+
+        public Dictionary<string, string> Environment { get; } = new Dictionary<string, string>();
+
+        public event DataReceivedEventHandler? ErrorDataReceived;
+
+        public event DataReceivedEventHandler? OutputDataReceived;
+
+        public string? WorkingDirectory { get; set; }
+
+        public Command(string command, string label="")
+        {
+            _command = command;
+            _label = label;
+        }
+
+        public Command WithWorkingDirectory(string dir)
+        {
+            WorkingDirectory = dir;
+            return this;
+        }
+
+        public Command WithEnvironmentVariable(string key, string value)
+        {
+            Environment[key] = value;
+            return this;
+        }
+
+        public Command WithEnvironmentVariables(IDictionary<string, string>? extraEnvVars)
+        {
+            if (extraEnvVars != null)
+            {
+                foreach ((string key, string value) in extraEnvVars)
+                    Environment[key] = value;
+            }
+
+            return this;
+        }
+
+        public virtual CommandResult Execute(string args = "")
```

Comment:
The argument construction is a bit confusing, maybe make an Arguments { get; } List<string> and make this. (I have a personal aversion to a single string as args).
```suggestion
        public virtual CommandResult Execute(string[] addtionalArguments)
```